addons/binding/org.openhab.binding.victronenergyvrm/src/main/java/org/openhab/binding/victronenergyvrm/handler/VictronEnergyVRMHandler.java

```diff
@@ -0,0 +1,146 @@
+
+        logger.debug("Try to authenticate against VRM API.");
+
+        VictronEnergyVRMAuth auth = new VictronEnergyVRMAuth(BASE_URL);
+
+        String token = auth.GetToken(username.toString(), password.toString());
+        this.token = token;
+
+        if (token != null) {
+            updateStatus(ThingStatus.ONLINE);
+            logger.debug("Login successfully");
+            // logger.debug("Here is the token: " + token);
+            pollingJob = scheduler.scheduleWithFixedDelay(this::updateData, 0, DEFAULT_REFRESH_RATE, TimeUnit.SECONDS);
+        }
+
+        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
+        // Long running initialization should be done asynchronously in background.
+
+        // Note: When initialization can NOT be done set the status with more details for further
+        // analysis. See also class ThingStatusDetail for all available status details.
+        // Add a description to give user information to understand why thing does not work
+        // as expected. E.g.
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+        // "Can not access device as username and/or password are invalid");
+    }
+
+    private void updateData() {
+        logger.debug("updateData");
+        final Object installId = getConfig().get(INSTALLID);
+        final Object instanceId = getConfig().get(INSTANCEID);
+        VictronEnergyVRMSolarChargerSummery ScS = new VictronEnergyVRMSolarChargerSummery(BASE_URL,
+                (BigDecimal) installId, (BigDecimal) instanceId);
+        ScS.loadData(this.token);
+        if (ScS.getSuccess()) {
+            if (!getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+                logger.debug("Set Errorcounter to 0");
+                errorcounter = 0;
+            }
+            logger.debug("yeah! Got Data from api VRMSolarChargerSummery");
+            updateState(CHANNEL_ScV, new DecimalType(ScS.getScV()));
+            updateState(CHANNEL_ScS, new StringType(ScS.getScS()));
+            updateState(CHANNEL_ScW, new DecimalType(ScS.getScW()));
+            updateState(CHANNEL_YT, new DecimalType(ScS.getYT()));
+            updateState(CHANNEL_YY, new DecimalType(ScS.getYY()));
+            updateState(CHANNEL_secondsAgo, new DecimalType(ScS.getSecondsAgo()));
+        } else {
+            // Wenn keine Daten geholt werden konnten und der Status nicht bereits schon "Offline" ist, setzt Status auf
+            // Offline.
+            if (!getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+                logger.warn("Couldn't get data from api VRMSolarChargerSummery");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            logger.debug("Increment Errorcounter.");
+            errorcounter++;
+            if (errorcounter < 3) {
+                this.initialize();

```

Comment:
This call will reassign the `pollingJob` in `initialize` to a new scheduled job and leak the previous assigned scheduled job, adding a new scheduled job each time this code is called.